include/rcpputils/thread_safety_annotations.h

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__THREAD_SAFETY_ANNOTATIONS_H_
+#define RCPPUTILS__THREAD_SAFETY_ANNOTATIONS_H_
+
+#include <mutex>
+
+// Enable thread safety attributes only with clang.
+// The attributes can be safely erased when compiling with other compilers.
+// Prefixing all macros to avoid potential conflict with other projects
+
+#if defined(__clang__) && (!defined(SWIG))
+#define RCPPUTILS_THREAD_ANNOTATION_ATTRIBUTE__(x)   __attribute__((x))
+#else
+#define RCPPUTILS_THREAD_ANNOTATION_ATTRIBUTE__(x)   // no-op
+#endif
+
+// libcxx does not define this operator, needed for negative capabilities
+// Here until someone has a better idea
+inline const std::mutex & operator!(const std::mutex & a)
+{
+  return a;
+}
+
+#define RCPPUTILS_CAPABILITY(x) \
```

Comment:
This is a bit bike-shed-y, but I think there will be lots of other utilities in the package at some point, so more namespacing than just `RCPPUTILS_` in these macros would be a good thing, imo. Maybe `RCPPUTILS_TSA_` (for Thread Safety Annotations) or `RCPPUTILS_TSAN_` (for Thread SANitizer).

Reading https://github.com/ros2/rmw_fastrtps/pull/259 I felt that things like `RCPPUTILS_REQUIRES` was pretty generic, and could use better namespacing.

Reply:
Makes sense - changed it